cicecore/cicedynB/general/ice_init.F90

```diff
@@ -1141,417 +1141,441 @@ fpfunc_andacc = 1      ! fixed point function for Anderson acceleration: 1: g(x)
          write(nu_diag,*) ' Grid, Discretization'
          write(nu_diag,*) '--------------------------------'
-         if (trim(grid_type) == 'rectangular') &
-         write(nu_diag,*)    'grid_type        = ', &
-                               trim(grid_type),': internally defined, rectangular grid'
-         if (trim(grid_type) == 'regional') &
-         write(nu_diag,*)    'grid_type        = ', &
-                               trim(grid_type),': user-defined, regional grid'
-         if (trim(grid_type) == 'displaced_pole') &
-         write(nu_diag,*)    'grid_type        = ', &
-                               trim(grid_type),': user-defined grid with rotated north pole'
-         if (trim(grid_type) == 'tripole') then
-         write(nu_diag,*)    'grid_type        = ', &
-                               trim(grid_type),': user-defined grid with northern hemisphere zipper'
-            if (trim(ns_boundary_type) == 'tripole') then
-               tmpstr2 = '  on U points (nodes)'
-            elseif (trim(ns_boundary_type) == 'tripoleT') then
-               tmpstr2 = '  on T points (cell centers)'
-            else
-               tmpstr2 = ' '
-            endif
-            write(nu_diag,*)    'ns_boundary_type = ', trim(ns_boundary_type),trim(tmpstr2)
-         endif
+         tmpstr2 = ' '
+         if (trim(grid_type) == 'rectangular')    tmpstr2 = ' : internally defined, rectangular grid'
+         if (trim(grid_type) == 'regional')       tmpstr2 = ' : user-defined, regional grid'
+         if (trim(grid_type) == 'displaced_pole') tmpstr2 = ' : user-defined grid with rotated north pole'
+         if (trim(grid_type) == 'tripole')        tmpstr2 = ' : user-defined grid with northern hemisphere zipper'
+         write(nu_diag,1030) ' grid_type        = ',trim(grid_type),trim(tmpstr2)
          if (trim(grid_type) /= 'rectangular') then
             if (use_bathymetry) then
-               tmpstr2 = ' bathymetric input data is used'
+               tmpstr2 = ' : bathymetric input data is used'
             else
-               tmpstr2 = ' bathymetric input data is not used'
+               tmpstr2 = ' : bathymetric input data is not used'
             endif
-            write(nu_diag,1012) ' use_bathymetry   = ', use_bathymetry,trim(tmpstr2)
-            write(nu_diag,*)    ' bathymetry_format= ', trim(bathymetry_format)
-         endif
-         write(nu_diag,1022) ' nilyr            = ', nilyr, ' number of ice layers (equal thickness)'
-         write(nu_diag,1022) ' nslyr            = ', nslyr, ' number of snow layers (equal thickness)'
-         write(nu_diag,1022) ' nblyr            = ', nblyr, ' number of bio layers (equal thickness)'
+            write(nu_diag,1010) ' use_bathymetry   = ', use_bathymetry,trim(tmpstr2)
+            write(nu_diag,1030) ' bathymetry_format= ', trim(bathymetry_format)
+         else  ! rectangular
+            write(nu_diag,1010) ' close_boundaries = ', close_boundaries
```

Comment:
I don't think we should write out close_boundaries, since it doesn't work.  It's one of those things that's hanging around in the code, waiting to be fixed.  The only way to close boundaries correctly is with a land mask.

Reply:
I understand now.  Should we add an abort if close_boundaries is true?

Reply:
Does the code not have anything in there now, to change the setting and write a warning, or to abort?  I'm fine with aborting with a message that it's not fully implemented yet, use a land mask instead.

Reply:
I can add an abort but box2001 and boxslotcyl both set close_boundaries to true.  Should this be changed to false?  I assume this will change answers?

Reply:
If those are using a land mask around the edges, then the answers shouldn't change if close_boundaries=F.  
If not:  @JFLemieux73 @phil-blain have either of you looked closely at what's happening near the boundaries, especially in the corners?  That's where I had problems.

Reply:
I have not.

Reply:
I'm running some tests now, turning off close_boundaries in the box test to see if it's bit-for-bit.  If it is, I think we can just set close_boundaries to false for those tests.  Will let you know what happens.

Reply:
Looks like switching to close_boundaries for the box2001 and boxslotcyl is NOT bit-for-bit.  How should we proceed?

Reply:
I'm sorry I didn't catch this earlier.  For now, leave the tests with the current settings so that the results don't change.  I'd take the close_boundaries setting out of the diagnostics output, the default ice_in file, and the namelist documentation (so it's somewhat hidden), leaving it as a namelist option in the code for backward compatibility, but we don't have to do this if you all think that's a bad idea.  It's okay with me to leave all of that in there but treat it as a known bug for this release (don't we have a list of known bugs in the docs somewhere?).  

For the future, please create an issue to fix the box tests so that the land mask is used rather than this flag, and/or to thoroughly test and perhaps fix close_boundaries.  I spent many weeks (months?) trying to figure that out and finally gave up, so I'm not optimistic.  At the very least, we need to understand why changing close_boundaries changes the answers when there's already a land mask present around the boundaries.  Maybe it needs more rows of land cells?  I have a vague memory that 2 rows were necessary.  I'll look for my notes on this the next time I go in to my office.

Reply:
OK, I have removed close_boundaries from the user guide and the verbose output.  We have to keep it in the default ice_in namelist because of how the scripts work.  For that value to be modified by the box tests, it has to exist in the default namelist, sorry about that.  An issue, #541, has been created for future reference.